syntax = "proto3";

package deepface.api.proto;

option go_package = "github.com/serengil/deepface/api/proto";

enum Models {
    VGG_FACE = 0;
    FACENET = 1;
    FACENET512 = 2;
    OPENFACE = 3;
    DEEPFACE = 4;
    DEEPID = 5;
    ARCFACE = 6;
    DLIB_MODEL = 7;
    SFACE = 8;
    GHOSTFACENET = 9;
    BUFFALO_L = 10;
}

enum Detectors {
    OPENCV = 0;
    SSD = 1;
    DLIB = 2;
    MTCNN = 3;
    FASTMTCNN = 4;
    RETINAFACE = 5;
    MEDIAPIPE = 6;
    YOLOV8 = 7;
    YOLOV11S = 8;
    YOLOV11N = 9;
    YOLOV11M = 10;
    YUNET = 11;
    CENTERFACE = 12;
}

enum DistanceMetrics {
    COSINE = 0;
    EUCLIDEAN = 1;
    EUCLIDEAN_L2 = 2;
    ANGULAR = 3;
}

message FacialArea {
    repeated uint32 left_eye = 1;
    repeated uint32 right_eye = 2;
    repeated uint32 mouth_left = 3;
    repeated uint32 mouth_right = 4;
    repeated uint32 nose = 5;
    uint32 h = 6;
    uint32 w = 7;
    uint32 x = 8;
    uint32 y = 9;
}